pkg/kms/service.go

```diff
@@ -165,13 +180,26 @@ func (s *service) VerifyMAC(keystoreID, keyID string, mac, data []byte) error {
 }
 
 func (s *service) getKeyHandle(keystoreID, keyID string) (interface{}, error) {
+	if err := s.checkKey(keystoreID, keyID); err != nil {
```

Comment:
return key in checkKey and remove Get() call below (to avoid calling Get() twice)

Reply:
keystore is different than keymanager... it's good